contracts/MarketRouter.sol

```diff
@@ -42,7 +44,7 @@ contract MarketRouter is CloberMarketSwapCallbackReceiver, CloberRouter {
         bytes calldata data
     ) external payable {
         // check if caller is registered market
-        if (_factory.getMarketHost(msg.sender) == address(0)) {
+        if (!_previousMarkets[msg.sender] && _factory.getMarketHost(msg.sender) == address(0)) {
```

Comment:
How about making this condition a public-view function named `registeredMarket`?